app/src/main/java/com/courseproject/tindar/entities/Account.java

```diff
@@ -0,0 +1,311 @@
+package com.courseproject.tindar.entities;
+
+import android.location.Location;
+import java.util.ArrayList;
+import java.util.Date;
+
+@SuppressWarnings("FieldMayBeFinal")
+
+public class Account {
+    private boolean isActiveStatus;
+    private final String id;
+    private String email;
+    private String password;
+    private String displayName;
+    private String firstName;
+    private String lastName;
+    private String aboutMe;
+    private String profilePictureLink;
+    private ArrayList<String> matchList;
+    private ArrayList<String> likeList;
+    private ArrayList<String> blockList;
+    private ArrayList<String> recentlyViewed;
+    private Date birthdate;
+    private Location location;
+    private Gender gender;
+    private Filters filters;
+    /**
+     * Initializes an Account objectwhich is used to store information
+     * about a user.
+     *
+     * @param id the back-end user id associated with this account
+     * @param email the email associated with this account
+     * @param password the password used to log in to this account
+     * @param display the username chosen to represent this account
+     * @param first the account owner's first name
+     * @param last the account owner's last name
+     */
+    Account(String id, String email, String password,
+                   String display, String first, String last) {
+        this.isActiveStatus = true;
+        this.id = id;
+        this.email = email;
+        this.password = password;
+        this.displayName = display;
+        this.firstName = first;
+        this.lastName = last;
+        this.aboutMe = "";
+        this.profilePictureLink = "";
+        this.matchList = new ArrayList<>();
+        this.likeList = new ArrayList<>();
+        this.blockList = new ArrayList<>();
+        this.recentlyViewed = new ArrayList<>();
+        this.birthdate = null;
+        this.location = null;
+        this.gender = null;
+        this.filters = null;
+    }
+    /** Returns the status of an account in the form of a String.
+     *
+     * @return the status of the account
+     */
+    String getAccountStatus() {
+        if (this.isActiveStatus) {
+            return "Active";
+        }
+        else {
+            return "Suspended";
+        }
+    }
+    /** Flips the account status from Active to Suspendedor vice versa.
+     *
+     */
+    void setAccountStatus() {
```

Comment:
I believe it would be more clear to pass the argument...like setAccountStatus(boolean isActive). The current implementation looks more like toggle than setting.